controllers/kubeadmcontrolplane_controller.go

```diff
@@ -201,6 +213,17 @@ func (r *KubeadmControlPlaneReconciler) validateStackedEtcd(kcp *controlplanev1.
 	return nil
 }
 
+func (r *KubeadmControlPlaneReconciler) pauseMachineHealthCheck(ctx context.Context, mhc *clusterv1.MachineHealthCheck, kcpName string) (ctrl.Result, error) {
+	if err := r.client.Get(ctx, GetNamespacedNameType(cpMachineHealthCheckName(kcpName), constants.EksaSystemNamespace), mhc); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		return ctrl.Result{}, err
+	}
+	annotations.AddAnnotations(mhc, map[string]string{machineHealthCheckPausedAnnotation: "true"})
```

Comment:
This is just adding the annotation to the local mhc object right? Don't you have to write the mhc back to cluster to ensure CAPI recognizes it as paused as well?